src/AnalyzeInterface.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Drupal\analyze;
+
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\Core\Url;
+
+/**
+ * Interface for analyze plugins.
+ */
+interface AnalyzeInterface {
+
+  /**
+   * Returns the translated plugin label.
+   */
+  public function label(): string;
+
+  /**
+   * Render a summary report for an entity.
+   *
+   * @param \Drupal\Core\Entity\EntityInterface $entity
+   *   The entity to render the summary for.
+   *
+   * @return mixed[]
+   *   A render array to include on the summary page.
+   */
+  public function renderSummary(EntityInterface $entity): array;
+
+  /**
+   * Render a full report for an entity.
+   *
+   * @param \Drupal\Core\Entity\EntityInterface $entity
+   *    The entity to render the summary for.
+   *
+   * @return mixed[]
+   *    A render array to include in the full report page.
+   */
+  public function renderFullReport(EntityInterface $entity): array;
```

Comment:
Should this be a nullable return? I guess you can return an empty array instead? Or it could be a supplementary and optional interface?

Reply:
The base method returns an empty array, so it would only return a NULL value if someone overrode the base method and then decided to return a NULL against the method's documentation. So I'd say no.

Reply:
Happy to leave as is